akka-http-marshallers-java/akka-http-jackson/src/main/java/akka/http/javadsl/marshallers/jackson/Jackson.java

```diff
@@ -35,23 +38,31 @@
   public static <T> Unmarshaller<ByteString, T> byteStringUnmarshaller(Class<T> expectedType) {
     return byteStringUnmarshaller(defaultObjectMapper, expectedType);
   }
-  
+
   public static <T> Unmarshaller<HttpEntity, T> unmarshaller(Class<T> expectedType) {
     return unmarshaller(defaultObjectMapper, expectedType);
   }
 
   public static <T> Unmarshaller<HttpEntity, T> unmarshaller(ObjectMapper mapper, Class<T> expectedType) {
-    return Unmarshaller.forMediaType(MediaTypes.APPLICATION_JSON, Unmarshaller.entityToString())
-                       .thenApply(s -> fromJSON(mapper, s, expectedType));
-  }
-  
+    return Unmarshaller
+      .forMediaType(MediaTypes.APPLICATION_JSON, Unmarshaller.entityToString())
+      .thenApply(s -> fromJSON(mapper, s, expectedType));
+    }
+
   public static <T> Unmarshaller<ByteString, T> byteStringUnmarshaller(ObjectMapper mapper, Class<T> expectedType) {
     return Unmarshaller.sync(s -> fromJSON(mapper, s.utf8String(), expectedType));
   }
 
   private static String toJSON(ObjectMapper mapper, Object object) {
     try {
-      return mapper.writeValueAsString(object);
+      final String candidate = mapper.writeValueAsString(object);
+
+      // Jackson (if root unwrapped) pushes `null` String into output instead of empty String.
+      if ("null".equals(candidate)) {
```

Comment:
Yeah, you are right. It will work for null objects of all types as jackson renders empty root to `"null"`, but indeed `"null"` will be rendered incorrectly.